tests/functional/search-auth.js

```diff
@@ -198,6 +198,48 @@ registerSuite("Search (auth)", {
         .end();
     },
 
+    "Label have correct URL after search without results"() {
+      // load up a garbage search, so we can more easily detect when
+      // the search values we want are loaded.
+      return FunctionalHelpers.openPage(
+        this,
+        url(
+          "/issues",
+          "?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc"
+        ),
+        ".grid"
+      )
+        .findByCssSelector("#js-SearchForm-input")
+        .clearValue()
+        .click()
+        .type("noExpectedResult123")
+        .submit()
+        .getProperty("value")
+        .then(function(text) {
+          assert.equal(
+            text,
+            "noExpectedResult123",
+            "The q param populated the search input."
```

Comment:
This assertion description doesn't match what the test is doing -- it's not looking at `q`. But that said, what is the point of this part of the test? You're entering text into the input and making sure that it's still there? Is there a situation where it wouldn't be?

Reply:
I had issues with the input and got in some cases an `undefined`, but this is probably worth a separate test. Will file a follow up issue.

Reply:
I can't reproduce right now. If it shows up again, I will add tests for that.